frontend/src/components/Search.js

```diff
@@ -0,0 +1,147 @@
+import { useState, useEffect } from "react";
+//style
+import styled from "styled-components";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+//redux
+import { useDispatch } from "react-redux";
+import { loadProductsData } from "../actions/productsAction";
+import { loadQuery } from "../actions/queryAction";
+
+const Search = () => {
+  //redux
+  const dispatch = useDispatch();
+  //state
+  const [input, setInput] = useState("");
+  const [price, setPrice] = useState("");
+  const [category, setCategory] = useState("");
+  const [query, setQuery] = useState("");
+
+  //event handlers
+  const handleInput = (e) => {
+    setInput(e.target.value);
+  };
+
+  const handlePriceClick = (e) => {
+    setPrice(e.target.value);
+  };
+
+  const handleCategoryClick = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSearchEvent = () => {
+    if (input && !price && !category) {
+      setQuery(`?query=${input}`);
+    } else if (price && !category && !input) {
+      setQuery(`?price=${price}`);
+    } else if (category && !input && !price) {
+      setQuery(`?category=${category}`);
+    } else if (input && category) {
+      if (price) {
+        return setQuery(`?query=${input}&category=${category}&price=${price}`);
+      }
+      setQuery(`?query=${input}&category=${category}`);
+    } else if (input && price) {
+      setQuery(`?query=${input}&price=${price}`);
+    } else if (price && category) {
+      setQuery(`?category=${category}&price=${price}`);
+    }
+  };
+
+  const handleKeyPress = (e) => {
+    if (e.code === "Enter") {
+      handleSearchEvent();
+    }
+  };
+
+  useEffect(() => {
+    if (query) {
+      dispatch(loadProductsData(query));
+      dispatch(loadQuery(query));
```

Comment:
The fact that you're dispatching `loadQuery` _after_ you dispatch `loadProductsData` is an indication that you really don't need `loadQuery` or that you should move some of the logic that is in your component here to a reducer.

Also, I'd say the naming of these two (using "load") is confusing because one does a fetch for data while the other sets the query.